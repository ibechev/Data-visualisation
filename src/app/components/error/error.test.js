import React from "react";
import { mount } from "enzyme";
import Error from "./Error";
import renderer from "react-test-renderer";
import { propTypeErrors } from "../../../../tests/config/testUtils";

describe("Expect <Error />", () => {
  const defaultProps = {
    errorText: "Error text"
  };

  it("to renderCorrectly", () => {
    const wrapper = renderer.create(<Error {...defaultProps} />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });

  it("to not throw error with correct props", () => {
    propTypeErrors(<Error {...defaultProps} />, defaultProps);
  });

  it("to display prop 'errorText' in <p class='error' />", () => {
    const wrapper = mount(<Error {...defaultProps} />);
    expect(wrapper.find("p.error").text()).toEqual(defaultProps.errorText);
  });
});
