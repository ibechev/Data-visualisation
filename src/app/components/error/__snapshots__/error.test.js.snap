// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expect <Error /> to renderCorrectly 1`] = `
<p
  className="error"
>
  Error text
</p>
`;
